### ISSUE TITLE
Error connect by key_file

### ISSUE THREAD

**Issue by high-v0ltage** [ktbyers/netmiko#827]
Key file is correct, I can connect using bash.
May be I doing something wrong?
netmiko 2.1.1
paramiko 2.4.1
Python 3.5.3
SMP Debian 4.9.65-3+deb9u2 (2018-01-04)
```
teno1 = {
    ...:     'device_type': 'linux',
    ...:     'ip': 'teno1',
    ...:     'username': 'root',
    ...:     'use_keys': 'True',
    ...:     'key_file': '/root/.ssh/id_rsa.pub',
    ...:     'allow_agent': 'True',
    ...:     'ssh_strict':'True',
    ...:     'system_host_keys': '/root/.ssh/known_hosts',
    ...:     'session_timeout': '1',
    ...:     #'system_host_keys': 'True',
    ...:     }
    ...: 
    ...: net_connect = ConnectHandler(**teno1)

```
```
`---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-15-ac47b3e90aca> in <module>()
     12     }
     13 
---> 14 net_connect = ConnectHandler(**teno1)

/usr/local/lib/python3.5/dist-packages/netmiko/ssh_dispatcher.py in ConnectHandler(*args, **kwargs)
    176                          'currently supported platforms are: {}'.format(platforms_str))
    177     ConnectionClass = ssh_dispatcher(kwargs['device_type'])
--> 178     return ConnectionClass(*args, **kwargs)
    179 
    180 

/usr/local/lib/python3.5/dist-packages/netmiko/base_connection.py in __init__(self, ip, host, username, password, secret, port, device_type, verbose, global_delay_factor, use_keys, key_file, allow_agent, ssh_strict, system_host_keys, alt_host_keys, alt_key_file, ssh_config_file, timeout, session_timeout, blocking_timeout, keepalive, default_enter, response_return, serial_settings)
    206             self._modify_connection_params()
    207             self.establish_connection()
--> 208             self.session_preparation()
    209 
    210     def __enter__(self):

/usr/local/lib/python3.5/dist-packages/netmiko/linux/linux_ssh.py in session_preparation(self)
     15         """Prepare the session after the connection has been established."""
     16         self.ansi_escape_codes = True
---> 17         return super(LinuxSSH, self).session_preparation()
     18 
     19     def _enter_shell(self):

/usr/local/lib/python3.5/dist-packages/netmiko/base_connection.py in session_preparation(self)
    578         """
    579         self._test_channel_read()
--> 580         self.set_base_prompt()
    581         self.disable_paging()
    582         self.set_terminal_width()

/usr/local/lib/python3.5/dist-packages/netmiko/linux/linux_ssh.py in set_base_prompt(self, pri_prompt_terminator, alt_prompt_terminator, delay_factor)
     35             pri_prompt_terminator=pri_prompt_terminator,
     36             alt_prompt_terminator=alt_prompt_terminator,
---> 37             delay_factor=delay_factor)
     38 
     39     def send_config_set(self, config_commands=None, exit_config_mode=True, **kwargs):

/usr/local/lib/python3.5/dist-packages/netmiko/base_connection.py in set_base_prompt(self, pri_prompt_terminator, alt_prompt_terminator, delay_factor)
    854         prompt = self.find_prompt(delay_factor=delay_factor)
    855         if not prompt[-1] in (pri_prompt_terminator, alt_prompt_terminator):
--> 856             raise ValueError("Router prompt not found: {0}".format(repr(prompt)))
    857         # Strip off trailing terminator
    858         self.base_prompt = prompt[:-1]

ValueError: Router prompt not found: '\x1b[1;31m[05:29]\x1b[1;37mroot@teno1:~#\x1b[0m'
```

**Comment by ktbyers** [comment]
@high-v0ltage You have ANSI escape codes in the output (possibly from the prompt being colored):

```
'\x1b[1;31m[05:29]\x1b[1;37mroot@teno1:~#\x1b[0m'
```

Probably that last one needs stripped: `\x1b[0m` which probably needs updated in the Netmiko strip_ansi_escape_codes method (in base_connection.py).

Netmiko is probably not the best choice for automating to Linux devices (as that is not what its main intent is).

**Comment by carlniger** [comment]
possibly fixed by #1196 
closing as this is quite old. feel free to reopen if needed!
